django/db/models/expressions.py

```diff
@@ -412,6 +432,12 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
     def refs_aggregate(self, existing_aggregates):
         return refs_aggregate(self.name.split(LOOKUP_SEP), existing_aggregates)
 
+    def asc(self):
```

Comment:
I wonder what is the use case of Resolvable that doesn't implement Combinable (btw should we just rename CombinableMixin to Combinable independent of other needed changes?)

I don't object making F an Expression subclass, but even in that case I think F shouldn't resolve to F-instance. So, resolve_expression would still work as it does now.